apps/frontend/static_src/js/main.js

```diff
@@ -1 +1,70 @@
 import { Dropdown } from 'bootstrap';
+
+const searchInput = document.querySelector('[data-search-input]')
+const searchIconButton = document.querySelector('[data-search-icon-button]')
+
+const onSearchInputChange = async (event) => {
+    const query = event.target.value
+    try {
+        const res = await fetch(window.location.origin + '/search_json/?query=' + query)
+        const data = await res.json()
+        injectResultsInHTML(data)
+    } catch(err) {
+        console.log(err)
+        return {"error": err}
+    }
+}
+searchInput.addEventListener("keyup", (event) => {
+    setTimeout(() => {
+        onSearchInputChange(event)
+    }, 1000)
+})
+
+
+const removeExistingChildren = (parent) => {
+    const children = [...parent.children]
+    for (const child of children) {
+        child.remove()
+    }
+}
+searchIconButton.addEventListener("click", () => {
+    const resultsDiv = document.querySelector('[data-results]')
+    const resultsCountContainer = document.querySelector('[data-results-count-container]')
+
+    removeExistingChildren(resultsDiv)
+    removeExistingChildren(resultsCountContainer)
+})
+
+const injectResultsInHTML = (results) => {
+    const resultsDiv = document.querySelector('[data-results]')
+    const resultsCountContainer = document.querySelector('[data-results-count-container]')
+
+    removeExistingChildren(resultsDiv)
+    removeExistingChildren(resultsCountContainer)
+
+    const resultsCountDiv = document.createElement("div")
+    const resultsCountData = document.createTextNode(`${results.length} ${results.length == 1 ? 'result' : 'results'} found`)
+    resultsCountDiv.appendChild(resultsCountData)
+    resultsCountDiv.classList.add('m-3', 'mx-5')
+    resultsCountContainer.appendChild(resultsCountDiv)
+
+    for(const result of results) {
+        const resultDiv = document.createElement("div")
+        const resultLink = document.createElement("a")
+        const resultDescription = document.createElement("div")
+        const resultParentSection = document.createElement("div")
+        resultLink.appendChild(document.createTextNode(result.title))
```

Comment:
A bit simpler for a case like this (and below):

```suggestion
        resultLink.innerText = result.title
```